### PR TITLE
Fix query parameter

### PR DESCRIPTION
**Pull request by fritzlitester**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{{#isFormParam}}{{^isFile}}@ApiParam(value = "{{{description}}}"{{#required}}, required=true{{/required}}{{#allowableValues}}, allowableValues="{{#values}}{{{.}}}{{^-last}}, {{/-last}}{{#-last}}{{/-last}}{{/values}}"{{/allowableValues}}{{^isContainer}}{{#defaultValue}}, defaultValue={{^isString}}"{{/isString}}{{#isString}}{{#isEnum}}"{{/isEnum}}{{/isString}}{{{defaultValue}}}{{^isString}}"{{/isString}}{{#isString}}{{#isEnum}}"{{/isEnum}}{{/isString}}{{/defaultValue}}{{/isContainer}}) @RequestPart(value="{{baseName}}"{{#required}}, required=true{{/required}}{{^required}}, required=false{{/required}})  {{{dataType}}} {{paramName}}{{/isFile}}{{#isFile}}@ApiParam(value = "{{{description}}}") {{#useBeanValidation}}@Valid{{/useBeanValidation}} @RequestPart(value = "{{baseName}}") {{#isListContainer}}List<{{/isListContainer}}MultipartFile{{#isListContainer}}>{{/isListContainer}} {{baseName}}{{/isFile}}{{/isFormParam}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache]: Please remove the EOL character.

File: modules/openapi-generator/src/main/resources/JavaSpring/queryParams.mustache [hunk 2]
```diff
@@ -1 +1 @@
...
...{/-last}}{{#-last}}{{/-last}}{{/values}}"{{/allowableValues}}{{^isContainer}}{{#defaultValue}}, defaultValue = {{^isString}}"{{/isString}}{{#isString}}{{#isEnum}}"{{/isEnum}}{{/isString}}{{{defaultValue}}}{{^isString}}"{{/isString}}{{#isString}}{{#isEnum}}"{{/isEnum}}{{/isString}}{{/defaultValue}}{{/isContainer}}) {{#useBeanValidation}}@Valid{{/useBeanValidation}}{{^isModel}} @RequestParam(value = {{#isMapContainer}}""{{/isMapContainer}}{{^isMapContainer}}"{{baseName}}"{{/isMapContainer}}{{#required}}, required = true{{/required}}{{^required}}, required = false{{/required}}{{^isContainer}}{{#defaultValue}}, defaultValue={{^isString}}"{{/isString}}{{#isString}}{{#isEnum}}"{{/isEnum}}{{/isString}}{{{defaultValue}}}{{^isString}}"{{/isString}}{{#isString}}{{#isEnum}}"{{/isEnum}}{{/isString}}{{/defaultValue}}{{/isContainer}}){{/isModel}}{{#isDate}} @org.springframework.format.annotation.DateTimeFormat(iso = org.springframework.format.annotation.DateTimeFormat.ISO.DATE){{/isDate}}{{#isDateTime}} @org.springframework.format.annotation.DateTimeFormat(iso = org.springframework.format.annotation.DateTimeFormat.ISO.DATE_TIME){{/isDateTime}} {{>optionalDataType}} {{paramName}}{{/isQueryParam}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaSpring/queryParams.mustache]: Please remove the EOL character

File: modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache (hunk 1 above)
Comment (fritzlitester) [reply modules/openapi-generator/src/main/resources/JavaSpring/formParams.mustache]: Tried to remove the EOL, not sure if it is correct now?

File: modules/openapi-generator/src/main/resources/JavaSpring/queryParams.mustache (hunk 2 above)
Comment (fritzlitester) [reply modules/openapi-generator/src/main/resources/JavaSpring/queryParams.mustache]: Tried to remove the EOL, not sure if it is correct now?

File: samples/server/petstore/springboot/pom.xml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>org.openapitools</groupId>
-    <artifactId>springboot</artifactId>
+    <artifactId>openapi-spring</artifactId>
     <packaging>jar</packaging>
-    <name>springboot</name>
-    <version>1.0.0-SNAPSHOT</version>
+    <name>openapi-spring</name>
+    <version>1.0.0</version>
```
Comment (wing328) [inline samples/server/petstore/springboot/pom.xml]: The change results in the following errors reported by the CI:
```
Running node 1 to test 'samples.circleci' defined in pom.xml ...
openjdk version "1.8.0_141"
OpenJDK Runtime Environment (build 1.8.0_141-8u141-b15-3~14.04-b15)
OpenJDK 64-Bit Server VM (build 25.141-b15, mixed mode)
[ERROR] Project 'org.openapitools:openapi-spring:1.0.0' is duplicated in the reactor -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DuplicateProjectException
```

Comment (fritzlitester) [reply samples/server/petstore/springboot/pom.xml]: This was auto generated. Did an undo manually now (don't know if this is the solution)

Comment (wing328) [reply samples/server/petstore/springboot/pom.xml]: Did you remove the file before regenerating the samples? (you shouldn't need to remove the samples manually)

Comment (fritzlitester) [reply samples/server/petstore/springboot/pom.xml]: No, I only modified the two mustache templates manually. The rest of the modifications must have been done by tests/regeneratiing the samples or probably the development environment.

Comment (fritzlitester) [reply samples/server/petstore/springboot/pom.xml]: Could the problem be, that I'm using Windows and there is no appropriate script to update the petstore sample for windows? Which scripts do I need to execute for Windows and my changes in the JavaSpring template such that the ci tests do not fail any more?

Comment (wing328) [reply samples/server/petstore/springboot/pom.xml]: ah.. likely. let me try to update the samples for you.

Comment (fritzlitester) [reply samples/server/petstore/springboot/pom.xml]: This would be nice.

**Comment by wing328** [comment]
@fritzlitester thanks for the PR, which has been included in the 4.3.0 release: https://twitter.com/oas_generator/status/1243455743937789952

**Comment by RaduFurnea** [comment]
Hello. Maybe I'm missing something, but why would you not allow @RequestParam in multipart anymore? This breaks functionality for people that have any other types apart from String and MultipartFile in their parameters, due to the converters used by Spring, further explained [here](https://github.com/spring-projects/spring-boot/issues/8441).

**Comment by trandersen-ufst** [comment]
@RaduFurnea This was fixed in https://github.com/OpenAPITools/openapi-generator/commit/33b89148e562fc2d5acf60a56719c46ec4f631e8 which is in 6.0.0-beta

**Pull request merged**